docs: add log signal release note and update docs (#10126)
determined-ai/determined: docs/release-notes/log-signal.rst

:orphan:

**New Features**

> -  Experiments: ``log_policies`` now have a ``name`` field. When a log policy matches, its name will

Reviewer: ```suggestion
-  Experiments: Add a ``name`` field to ``log_policies``. When a log policy matches, its name 
```

>    be displayed as a label in the WebUI, allowing for easy identification of specific issues during

Reviewer: ```suggestion
   shows as a label in the WebUI, making it easy to spot specific issues during 
```

>    a run. These labels will appear in both the run table and run detail views.

Reviewer: ```suggestion
a run. Labels appear in both the run table and run detail views.
```

>    It has a new format. ``name`` is required, and ``action`` should be a plain string. For more

Reviewer: ```suggestion
   In addition, there is a new format. ``name`` is required, and ``action`` is now a plain string. For more
```

:orphan:

**New Features**

-  Experiments: ``log_policies`` now have a ``name`` field. When a log policy matches, its name will
   be displayed as a label in the WebUI, allowing for easy identification of specific issues during
   a run. These labels will appear in both the run table and run detail views.

   It has a new format. ``name`` is required. ``pattern`` and ``action`` are optional. To make

>    things simpler, user no longer needs to specify the ``type`` field to set an action. For example:

Reviewer 2: > It has a new format.  `name` is required, and `action` should be a plain string.  See <link to docs> for more details.

Then skip the rest of the text.

The release note should be a brief note and interested users should go to the main docs.
